docs/guide/why.md

```diff
@@ -51,9 +51,9 @@ Ensuring optimal output and behavioral consistency between the dev server and th
 
 ## Why Not Bundle with esbuild?
 
-When Vite was first created, `esbuild` was not used for builds as features like code-splitting and CSS handling were still a work in progress, so Rollup was used instead as it is more mature and flexible in these regards.
+Vite's adoption of Rollup's plugin API and infrastructure heavily contributed to its success in the ecosystem, however, it isn't compatible with `esbuild`'s API. For the time being, we believe that Rollup offers a better performance-vs-flexibility tradeoff.
```

Comment:
One last suggestion only reordering the lines 👀
I'm fine with either though
```suggestion
Vite's current plugin API isn't compatible with using `esbuild` as a bundler. For the time being, we believe that Rollup offers a better performance-vs-flexibility tradeoff. Vite's adoption of Rollup's plugin API and infrastructure heavily contributed to its success in the ecosystem.
```

Reply:
It might be clearer to just say "better flexibility" than "better performance-vs-flexibility tradeoff" as the latter requires some reader knowledge of which one is faster and which one is more flexible to be able to interpret

Reply:
Is this more clear?

> Vite's current plugin API isn't compatible with using `esbuild` as a bundler. In spite of `esbuild` being faster, Vite's adoption of Rollup's flexible plugin API and infrastructure heavily contributed to its success in the ecosystem. For the time being, we believe that Rollup offers a better performance-vs-flexibility tradeoff.

Reply:
I like that too. Will change to that 👍 
```suggestion
Vite's current plugin API isn't compatible with using `esbuild` as a bundler. In spite of `esbuild` being faster, Vite's adoption of Rollup's flexible plugin API and infrastructure heavily contributed to its success in the ecosystem. For the time being, we believe that Rollup offers a better performance-vs-flexibility tradeoff.
```